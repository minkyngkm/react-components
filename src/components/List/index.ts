export { default } from "./List";
export type { Props as ListProps } from "./List";
