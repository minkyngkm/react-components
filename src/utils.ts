export const IS_DEV = process.env.NODE_ENV === "development";

/**
 * Find substring and wrap in <strong /> tag
 * @param {string} str - The string to search
 * @param {string} subString - The substring to find
 * @return {Obj} newStr - Object with text and match bool
 */
export const highlightSubString = (
  str?: string,
  subString?: string
): { text: string; match: boolean } => {
  if (typeof str !== "string" || typeof subString !== "string") {
    return {
      text: str || "",
      match: false,
    };
  }
  const caseInsensitiveRegex = new RegExp(subString, "gi");
  const newStr = str.replace(
    caseInsensitiveRegex,
    (match) => `<strong>${match}</strong>`
  );

  return {
    text: subString === "" ? str : newStr,
    match: newStr !== str,
  };
};
