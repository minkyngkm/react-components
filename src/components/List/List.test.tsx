import { shallow } from "enzyme";
import React from "react";

import { compareJSX } from "../../testing/utils";
import List from "./List";

describe("List ", () => {
  let items: string[];

  beforeEach(() => {
    items = ["test", "items", "here"];
  });

  it("renders", () => {
    const wrapper = shallow(<List items={items} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("can define items as JSX", () => {
    const wrapper = shallow(
      <List items={[<div>test</div>, <span>items</span>]} />
    );
    const item = wrapper.find(".p-list__item").first();
    compareJSX(item.children(), <div>test</div>);
  });

  it("can define items as objects", () => {
    const wrapper = shallow(
      <List items={[{ content: "test", className: "item-class" }]} />
    );
    const item = wrapper.find(".p-list__item").first();
    expect(item.prop("className").includes("item-class")).toBe(true);
  });

  it("can be divided", () => {
    const wrapper = shallow(<List divided={true} items={items} />);
    expect(wrapper.prop("className").includes("p-list--divided")).toBe(true);
  });

  it("can be inline", () => {
    const wrapper = shallow(<List inline={true} items={items} />);
    expect(wrapper.prop("className").includes("p-inline-list")).toBe(true);
    expect(wrapper.find(".p-inline-list__item").exists()).toBe(true);
  });

  it("can be inline middot", () => {
    const wrapper = shallow(<List middot={true} items={items} />);
    expect(wrapper.prop("className").includes("p-inline-list--middot")).toBe(
      true
    );
    expect(wrapper.find(".p-inline-list__item").exists()).toBe(true);
  });

  it("can be inline stretch", () => {
    const wrapper = shallow(<List stretch={true} items={items} />);
    expect(wrapper.prop("className").includes("p-inline-list--stretch")).toBe(
      true
    );
    expect(wrapper.find(".p-inline-list__item").exists()).toBe(true);
    expect(wrapper.prop("className").includes("p-list")).toBe(false);
  });

  it("can be split", () => {
    const wrapper = shallow(<List split={true} items={items} />);
    expect(wrapper.prop("className").includes("is-split")).toBe(true);
  });

  it("can be ticked", () => {
    const wrapper = shallow(<List ticked={true} items={items} />);
    expect(
      wrapper
        .find(".p-list__item")
        .first()
        .prop("className")
        .includes("is-ticked")
    ).toBe(true);
  });

  it("can be stepped", () => {
    const wrapper = shallow(
      <List
        stepped={true}
        items={[
          {
            title: "Test title",
            content: "test",
            className: "item-class",
            titleElement: "h1",
          },
        ]}
      />
    );
    expect(wrapper.prop("className").includes("p-stepped-list")).toBe(true);
    compareJSX(
      wrapper.find(".p-stepped-list__item").first(),
      <li className="item-class p-stepped-list__item">
        <h1 className="p-stepped-list__title">Test title</h1>
        <p className="p-stepped-list__content">test</p>
      </li>
    );
  });

  it("can be detailed stepped", () => {
    const wrapper = shallow(
      <List
        detailed={true}
        stepped={true}
        items={[
          {
            title: "Test title",
            content: "test",
            className: "item-class",
            titleElement: "h1",
          },
        ]}
      />
    );
    expect(wrapper.prop("className").includes("p-stepped-list--detailed")).toBe(
      true
    );
    expect(wrapper.find(".p-stepped-list__item").exists()).toBe(true);
  });

  it("can add additional classes", () => {
    const wrapper = shallow(<List className="extra-class" items={items} />);
    const className = wrapper.prop("className");
    expect(className.includes("p-list")).toBe(true);
    expect(className.includes("extra-class")).toBe(true);
  });
});
