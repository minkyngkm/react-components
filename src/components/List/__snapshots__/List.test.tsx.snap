// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List  renders 1`] = `
<ul
  className="p-list"
>
  <li
    className="p-list__item"
    key="0"
  >
    test
  </li>
  <li
    className="p-list__item"
    key="1"
  >
    items
  </li>
  <li
    className="p-list__item"
    key="2"
  >
    here
  </li>
</ul>
`;
